src/GDBDebugSession.ts

```
@@ -1792,6 +1806,65 @@ export class GDBDebugSession extends LoggingDebugSession {
         return Promise.resolve(variables);
     }
 
+    // Register view
+
+    private registerMap = new Map<string, number>();
+    private registerMapReverse = new Map<number, string>();
+    protected async handleVariableRequestRegister(ref: RegisterVariableReference): Promise<DebugProtocol.Variable[]> {
+        // initialize variables array and dereference the frame handle
+        const variables: DebugProtocol.Variable[] = [];
+        const frame = this.frameHandles.get(ref.frameHandle);
+        if (!frame) {
+            return Promise.resolve(variables);
+        }
+        
+        try {
+            if (this.registerMap.size === 0) {
+                ////// Refer to e2s plugin to port request code
+                const result_names = await mi.sendDataListRegisterNames(this.gdb, {});
+                let idx = 0;
+                const registerNames = result_names['register-names'];
+                for (const regs of registerNames){
+                    if(regs !== ''){
+                        this.registerMap.set(regs, idx);
+                        this.registerMapReverse.set(idx, regs);
+                    }
+                    idx++;
+                }      
+            }            
+        }
+        catch {
```

Don't discard the exception information, use it so that the source of the problem can be tracked down later.

---

Sorry but I cannot catch your idea here.
Could you please explain more?

---

Since "handleVariableRequestRegister" is called in "variablesRequest", and in this function, the error information is returned after handled "handleVariableRequestRegister". So can we remove the "catch (error)" here? How do you think?

---

That is a good idea.